### PR TITLE
DDF-3660 Added some unit tests

### PR DESCRIPTION
**Pull request by emmberk**

### CODE REVIEW

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/DeleteRequestImpl.java [hunk 1]
```diff
@@ -38,18 +38,29 @@
...
+   */
+  public DeleteRequestImpl(String id, Map<String, Serializable> properties) {
+    this(new String[] {id}, properties);
+  }
+
   /**
    * Instantiates a new DeleteRequestImpl with an array of {@link String} id to be deleted.
    *
-   * @param id1 - the id to be used for the delete operation
+   * @param ids - the id to be used for the delete operation
```
Comment (peterhuffer) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/DeleteRequestImpl.java]: ✏️ Nit: the second `ids`

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java [hunk 2]
```diff
@@ -232,6 +233,7 @@
    * @param name name of attribute
    * @return {@link Attribute} for given name, or <code>null</code> if not available
    */
+  @Nullable
```
Comment (peterhuffer) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java]: ✏️ Reminder to remove.

File: catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/HarvestedResourceTransformer.java [hunk 3]
```diff
@@ -84,36 +74,30 @@ public HarvestedResourceTransformer(
...
       IOUtils.copy(harvestedResource.getInputStream(), tfbos);
       final ByteSource byteSource = tfbos.asByteSource();
 
-      final MimeType mimeType =
-          Optional.ofNullable(harvestedResource.getMimeType())
-              .orElse(
-                  guessMimeTypeFor(
-                      byteSource.openStream(),
-                      FilenameUtils.getExtension(harvestedResource.getName())));
+      MimeType mimeType = harvestedResource.getMimeType();
```
Comment (peterhuffer) [inline catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/HarvestedResourceTransformer.java]: ❓ Why was this changed?

File: catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java [hunk 4]
```diff
@@ -54,7 +51,7 @@
 
   private final Sardine sardine = SardineFactory.begin();
 
-  private Map<String, Serializable> attributeOverrides = new HashMap<>();
+  private Map<String, List<String>> attributeOverrides = new HashMap<>();
```
Comment (peterhuffer) [inline catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java:54]: ✏️ Something I thought of right now is when these Harvesters eventually support ingesting the content into the content store (as a configurable option), we'll want to use the content frameworks method of attribute overrides.

File: platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeMapper.java [hunk 5]
```diff
@@ -53,6 +53,6 @@ public String getMimeTypeForFileExtension(String fileExtension)
    *     determine the mime type
    * @throws MimeTypeResolutionException if any problems encountered during mime type mapping
    */
-  public String guessMimeType(InputStream is, String fileExtension)
-      throws MimeTypeResolutionException;
+  @Nullable
```
Comment (peterhuffer) [inline platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeMapper.java]: May need to account for compatibility here.

File: platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeToTransformerMapper.java [hunk 6]
```diff
@@ -15,13 +15,14 @@
 
 import java.util.List;
 import javax.activation.MimeType;
+import javax.annotation.Nullable;
 
 /** @author Ashraf Barakat */
 public interface MimeTypeToTransformerMapper {
 
-  public static final String MIME_TYPE_KEY = "mime-type";
```
Comment (peterhuffer) [inline platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeToTransformerMapper.java]: Not sure if this changes compatibility.

**Review by peterhuffer — APPROVED** [review]

File: catalog/harvest/catalog-harvest-webdav/src/test/java/org/codice/ddf/catalog/harvest/webdav/DavAlterationObserverTest.java [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DavAlterationObserverTest {
```
Comment (emmberk) [inline catalog/harvest/catalog-harvest-webdav/src/test/java/org/codice/ddf/catalog/harvest/webdav/DavAlterationObserverTest.java:37]: This file was moved from https://github.com/codice/ddf/blob/3efc289a995afbddf1170c716233a34ee82eaab8/catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserverTest.java without any changes.

File: catalog/harvest/catalog-harvest-webdav/src/test/java/org/codice/ddf/catalog/harvest/webdav/DavEntryTest.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.Date;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DavEntryTest {
```
Comment (emmberk) [inline catalog/harvest/catalog-harvest-webdav/src/test/java/org/codice/ddf/catalog/harvest/webdav/DavEntryTest.java:34]: This file was moved from https://github.com/codice/ddf/blob/3efc289a995afbddf1170c716233a34ee82eaab8/catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/DavEntryTest.javawithout any changes.

File: platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeToTransformerMapper.java (hunk 6 above)
Comment (emmberk) [reply platform/mime/core/platform-mime-core-api/src/main/java/ddf/mime/MimeTypeToTransformerMapper.java]: I don't think it will. I think `public static final` is implied.

File: catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/HarvestedResourceTransformer.java [hunk 9]
```diff
@@ -84,32 +74,26 @@ public HarvestedResourceTransformer(
...
-      throws HarvestException {
+  Metacard transformHarvestedResource(
+      HarvestedResource harvestedResource, @Nullable String metacardId) throws HarvestException {
     try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
       IOUtils.copy(harvestedResource.getInputStream(), tfbos);
       final ByteSource byteSource = tfbos.asByteSource();
 
       final MimeType mimeType =
-          Optional.ofNullable(harvestedResource.getMimeType())
+          Optional.of(harvestedResource.getMimeType())
```
Comment (peterhuffer) [inline catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/HarvestedResourceTransformer.java]: May need to change back cause it looks like `getMimeType()` can return null. It's not documented in the `BinaryContent` interface, but the implementation allows it.

File: catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/MetacardOnlyAdaptor.java [hunk 10]
```diff
@@ -129,19 +129,20 @@ public void update(HarvestedResource resource, String updateId) throws HarvestEx
...
           }
+
+          return null;
         });
   }
 
   @Override
   public void delete(String id) {
-    DeleteRequest deleteRequest = new DeleteRequestImpl(id);
+    final DeleteRequest deleteRequest = new DeleteRequestImpl(id, getSecurityMap());
```
Comment (peterhuffer) [inline catalog/harvest/catalog-harvest-metacardadaptor/src/main/java/org/codice/ddf/catalog/harvest/adaptor/MetacardOnlyAdaptor.java:139]: This will probably need to be tested, but I believe there are issues passing an unmodifiable map into the Catalog framework.

File: catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java [hunk 11]
```diff
@@ -241,15 +244,22 @@ public void setAttributeOverrides(List<String> incomingAttrOverrides) {
...
+          Arrays.asList(newValue),
+          (oldValue, v) -> {
+            oldValue.add(newValue);
+            return oldValue;
+          });
     }
   }
 
-  public void setWebdavAddress(String webdavAddress) {
+  private void setWebdavAddress(String webdavAddress) {
```
Comment (peterhuffer) [inline catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java:258]: I think since these are setters they should be public for convention.

Comment (emmberk) [reply catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java:258]: The public setter is `updateCallback`. `updateCallback` is the only setter that should be used because it does some argument validation and also reinitializes the Harvester.
